### PR TITLE
Ma 104 show tenancy

### PR DESCRIPTION
**Pull request by elena-vi**
Added endpoint `GET 'api/v1/tenancies/:tenancy_ref'`
which returns the following: 
```json
{
    "id": 1,
    "tenancy_ref": "055593/01",
    "priority_band": "red",
    "priority_score": 21563,
    "created_at": "2019-04-02T03:26:35.000Z",
    "updated_at": "2019-09-16T16:48:29.410Z",
    "balance_contribution": "517.0",
    "days_in_arrears_contribution": "1689.0",
    "days_since_last_payment_contribution": "214725.0",
    "payment_amount_delta_contribution": "-900.0",
    "payment_date_delta_contribution": "30.0",
    "number_of_broken_agreements_contribution": "0.0",
    "active_agreement_contribution": null,
    "broken_court_order_contribution": null,
    "nosp_served_contribution": null,
    "active_nosp_contribution": null,
    "balance": "430.9",
    "days_in_arrears": 1126,
    "days_since_last_payment": 1227,
    "payment_amount_delta": "-900.0",
    "payment_date_delta": 6,
    "number_of_broken_agreements": 0,
    "active_agreement": false,
    "broken_court_order": false,
    "nosp_served": false,
    "active_nosp": false,
    "assigned_user_id": 1,
    "is_paused_until": null,
    "pause_reason": null,
    "pause_comment": null,
    "case_id": 7250,
    "assigned_user": {
        "id": 1,
        "provider_uid": "AIHAIEROUAWEB",
        "provider": "azureactivedirectory",
        "name": "Elena Vilimaite",
        "email": "Elena.Vilimaite@hackney.gov.uk",
        "first_name": "Elena",
        "last_name": "Vilimaite",
        "provider_permissions": "true",
        "role": "base_user"
    }
}
```

As part of the following ticket: https://hackney.atlassian.net/browse/MA-104?atlOrigin=eyJpIjoiYTgwOTY4ZDdlYmM4NDIzNWI0MGI0OGJjZGU3ZGU5YzEiLCJwIjoiaiJ9

### CODE REVIEW

File: spec/controllers/tenancies_controller_spec.rb [hunk 1]
```diff
@@ -98,4 +121,43 @@ class StubSqlPauseTenancyGateway
...
+      'payment_amount_delta' => -900,
+      'payment_date_delta' => 6,
+      'number_of_broken_agreements' => 0,
+      'active_agreement' => false,
+      'broken_court_order' => false,
+      'nosp_served' => false,
+      'active_nosp' => false,
+      'assigned_user' => {
+        user_id: 128,
+        name: 'George'
```
Comment (georgeschena) [inline spec/controllers/tenancies_controller_spec.rb]: 👍

File: spec/lib/hackney/income/sql_tenancy_case_gateway_spec.rb [hunk 2]
```diff
@@ -189,6 +189,29 @@
...
+        tenancy_1.save
+      end
+
+      it 'gets a case' do
+        subject.get_tenancy(
+          tenancy_ref: tenancy_1.tenancy_ref
+        )
+      end
+
+      it 'raises and error when not found' do
```
Comment (kjdchapman) [inline spec/lib/hackney/income/sql_tenancy_case_gateway_spec.rb]: nit: typo
```suggestion
      it 'raises an error when not found' do
```

File: lib/hackney/income/get_tenancy.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
+module Hackney
+  module Income
+    class GetTenancy
```
Comment (kjdchapman) [inline lib/hackney/income/get_tenancy.rb]: Is the naming convention to not explicitly call out these are `UseCases`? I've typically used modules to help people find these.

File: lib/hackney/income/sql_tenancy_case_gateway.rb [hunk 4]
```diff
@@ -30,6 +30,17 @@ def assign_to_next_available_user(tenancy:)
         tenancy.assigned_user_id
       end
 
+      def get_tenancy(tenancy_ref:)
```
[the remark this answers is not in the document]
Comment (georgeschena) [reply lib/hackney/income/sql_tenancy_case_gateway.rb]: 👍

**Pull request merged**